clients/vscode-hlasmplugin/src/hlasmDownloadCommands.ts

```diff
@@ -0,0 +1,697 @@
+            jobNo = jobNo / translationTable.length | 0;
+        }
+        if (jobnameSuffix.length > header.jobHeader.replCount)
+            jobnameSuffix = jobnameSuffix.slice(jobnameSuffix.length - header.jobHeader.replCount);
+        else if (jobnameSuffix.length < header.jobHeader.replCount)
+            jobnameSuffix = jobnameSuffix.padStart(header.jobHeader.replCount, '0');
+
+        return header.jobHeader.prefix + jobnameSuffix + header.jobHeader.suffix;
+    }
+
+}
+
+function generateJcl(jobNo: number, jobcard: ParsedJobHeader, datasetName: string): string {
+    return [
+        generateJobHeader(jobcard, jobNo),
+        "//AMATERSE EXEC PGM=AMATERSE,PARM=SPACK",
+        "//SYSPRINT DD DUMMY",
+        "//SYSIN    DD DUMMY",
+        "//SYSUT1   DD DISP=SHR,DSN=" + datasetName,
+        "//SYSUT2   DD DISP=(,PASS),DSN=&&TERSED,SPACE=(CYL,(10,10))",
+        "//*",
+        "//PRINTIT  EXEC PGM=IEBGENER",
+        "//SYSPRINT DD DUMMY",
+        "//SYSIN    DD DUMMY",
+        "//SYSUT1   DD DISP=OLD,DSN=&&TERSED",
+        "//SYSUT2   DD SYSOUT=*"
+    ].join(EOL)
+}
+
+function extractJobName(jcl: string): string {
+    return jcl.slice(2, jcl.indexOf(' '));
+}
+
+async function submitJobs(client: JobClient, jobcard: ParsedJobHeader, jobList: JobDetail[], progress: StageProgressReporter, checkCancel: () => void): Promise<SubmittedJob[]> {
+    let id = 0;
+
+    let result: SubmittedJob[] = [];
+
+    for (const e of jobList) {
+        checkCancel();
+        const jcl = generateJcl(id++, jobcard, e.dsn);
+        const jobname = extractJobName(jcl);
+
+        const jobid = await client.submitJcl(jcl);
+
+        result.push({ jobname: jobname, jobid: jobid, details: e, downloaded: false });
+        progress.stageCompleted();
+    }
+    return result;
+}
+
+function fixPath(p: string): string {
+    p = p.replace(/\\/g, '/');
+    while (p.endsWith('/')) // no lastIndexOfNot or trimEnd(x)?
+        p = p.slice(0, p.length - 1);
+    return p;
+}
+
+function getWasmRuntimeArgs(): Array<string> {
+    const v8Version = process && process.versions && process.versions.v8 || "1.0";
+    const v8Major = +v8Version.split(".")[0];
+    if (v8Major >= 10)
+        return [];
+    else
+        return [
+            '--experimental-wasm-eh'
+        ];
+}
+
+async function unterse(outDir: string): Promise<{ process: Promise<void>, input: Writable }> {
+    await fsp.mkdir(outDir, { recursive: true });
+    const unpacker = fork(
+        path.join(__dirname, '..', 'bin', 'terse'),
+        ["--op", "unpack", "--overwrite", "--copy-if-symlink-fails", "-o", outDir],
+        { execArgv: getWasmRuntimeArgs(), stdio: ['pipe', 'ignore', 'pipe', 'ipc'] }
+    );
+    const promise = new Promise<void>((resolve, reject) => {
+        unpacker.stderr!.on('data', (chunk) => console.log(chunk.toString()));
+        unpacker.on('exit', (code, signal) => {
+            if (signal)
```

Comment:
According to the [docs](https://nodejs.org/api/child_process.html#child_process_event_exit) one of the two arguments will always be "null". So the check should be `if (code === null)` ...`. But since signal codes are non-empty strings it will work the same this way.

Reply:
the process can fail either due to a signal, or due to a non-zero error code. `code === null` tells you basically nothing.

Reply:
what's your point?

Reply:
That according to the node docs the check for termination by signal should either be `if (code === null)` or `if (signal !== null)`.